sycl/test-e2e/DeviceDependencies/Inputs/a.cpp

```diff
@@ -1,11 +1,12 @@
+#define A_EXPORT
 #include "a.hpp"
 #include "b.hpp"
 #include <iostream>
 
-SYCL_EXTERNAL int levelA(int val) {
+A_DECLSPEC SYCL_EXTERNAL int levelA(int val) {
 #ifndef __SYCL_DEVICE_ONLY__
   std::cerr << "Host symbol used" << std::endl;
-  return 0;
+  val^=0x1234;
```

Comment:
Could you please clarify what this change on host code is necessary?

Reply:
The `val^=XXXX;` lines are necessary because otherwise the testcase will pass even without the `-fsycl-allow-device-dependencies` option.  Without `-fsycl-allow-device-dependencies` the SYCL RT will not find the symbols needed in device code, but the SYCL RT will find an implementation of the missing symbol produced in the host compilation flow (i.e. compilation flow where __SYCL_DEVICE_ONLY is not defined) and use that version.  @sergey-semenov can you review this comment.

Reply:
To clarify, that's how it would behave specifically with OpenCL CPU. Since the device code is executed in the same context as the host code there, it will happily use host definitions of a symbol if the device counterpart is unresolved. The reason  for the switch from "return 0" is that after dead code elimination there was no host dependency on the next DLL and so it was simply never loaded on Windows.

Reply:
> it will happily use host definitions of a symbol if the device counterpart is unresolved.

One question I have is why do we have this behavior?  CPU is supposed to simplify development for GPU.  This behavior will hide a problem when the device code is run on GPU.